client/packages/programs/src/JsonForms/components/EnrolmentId.tsx

```diff
@@ -33,7 +34,7 @@ const UIComponent = (props: ControlProps) => {
 
   const { core } = useJsonForms();
   const patientId = options?.patientIdField
-    ? extractProperty(core?.data, options?.patientIdField ?? '')
+    ? extractProperty(core?.data, options?.patientIdField ?? '') ?? '' // use empty/invalid id if field is not set
```

Comment:
there is a fallback parameter on extractProperty, i think this is equivalent:
`const patientId  = extractProperty(core?.data, options?.patientIdField ?? '',  config?.patientId)`.

Btw what is config ? Is it document ?

Reply:
Thanks using the fallback is much better.

The config is a custom object that is passed into the json forms and all controls in the json form can access it. The config contains common data such as storeId, patient, etc...

Thanks for bringing it up. There was an existing type for the config. I am now using it everywhere and added some missing fields... It should make it easier now to find where it comes from.